### PR TITLE
build(dependabot): automatic pull request creation and merge for component-adapter-interfaces

### PR DESCRIPTION
**Pull request by maloun96**
Automate updating the components interface dependency.

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-227495

### CODE REVIEW

**Comment by maloun96** [comment]
@lalli-flores

File: .github/workflows/automerge.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  pull_request:
+
+jobs:
+  worker:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: automerge
+        if: github.event.pull_request.user.login == 'dependabot[bot]'
+        uses: actions/github-script@0.2.0
```
Comment (lalli-flores) [inline .github/workflows/automerge.yml:13]: I'm pretty sure there's a newer version out?

**Review by lalli-flores — APPROVED** [review]
@maloun96 did you test this workflow?

File: .github/workflows/automerge.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: "Dependabot Automerge"
```
Comment (cipak) [inline .github/workflows/automerge.yml:1]: Dependabot Automerge Webex Dependency Updates

File: .github/workflows/automerge-dependabot-webex-dependency-updates.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    branches:
+      - master
+
+jobs:
+  dependency-automerge:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: automerge
+        if: github.event.pull_request.user.login == 'dependabot[bot]'
```
Comment (cipak) [inline .github/workflows/automerge-dependabot-webex-dependency-updates.yml:14]: won't this also merge other dependabot PRs?

File: .github/dependabot.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+# package ecosystems to update and where the package manifests are located.
+# Please see the documentation for all configuration options:
+# https://help.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+version: 2
+updates:
+  - package-ecosystem: "npm" # See documentation for possible values
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "daily"
+      time: "12:00"
```
[the remark this answers is not in the document]
Comment (maloun96) [reply .github/dependabot.yml]: We agree inside our team to put 9:00 AM which is better for us. What do you think?

File: .github/workflows/automerge-dependabot-webex-dependency-updates.yml (hunk 3 above)
Comment (maloun96) [reply .github/workflows/automerge-dependabot-webex-dependency-updates.yml:14]: It will

Comment (cipak) [reply .github/workflows/automerge-dependabot-webex-dependency-updates.yml:14]: @lalli-flores are you ok with merging the other dependabot PRs, for third party dependencies, or should we find a way to identify the webex PRs specifically?

**Review by lalli-flores — CHANGES_REQUESTED** [review]
I had sometime to investigate what alternatives we have to dependabot because I feel that the current setup could potentially harm more than help, plus with the limitations (scheduled once a day) is not as useful as intended.

So here's my proposed solution:

1. Github actions can be triggered remotely by using [their dispatch API](https://docs.github.com/en/rest/reference/actions#create-a-workflow-dispatch-event). What this means for us is that we can update the [interface's repo release workflow](https://github.com/webex/component-adapter-interfaces/blob/master/.circleci/config.yml#L83) and add a step after release to trigger Github actions in components and adapter repos via the API.
2. In this repo (and similarly in SDK adapter repo) we have a Github action that is listening for that dispatch event.
3. When the dispatch event listener in the action is triggered and checks out the repo
4. we can use [NPM's outdated](https://docs.npmjs.com/cli/v7/commands/npm-outdated) to check if the interface's package is out of date
```
npm outdated @webex/component-adapter-interfaces --json
```
4. If version is outdated, updated to `wanted` version
5. Commit update to master

**Comment by maloun96** [comment]
@lalli-flores This PR can be closed then. I created as you suggested. Check, please
https://github.com/webex/component-adapter-interfaces/pull/53
https://github.com/webex/components/pulls

**Comment by lalli-flores** [comment]
@maloun96 correct. I've left feedback on the new PRs.

**Pull request closed without merging**